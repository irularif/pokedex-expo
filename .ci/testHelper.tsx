import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import { render, fireEvent, act } from "@testing-library/react-native";
import { SafeAreaProvider } from "react-native-safe-area-context";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  const options = {
    wrapper: (props: any) => (
      <SafeAreaProvider>
        <View {...props} testID="wrapper" />
      </SafeAreaProvider>
    ),
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId("wrapper")!;
  return { wrapper, ...renderApi };
};
