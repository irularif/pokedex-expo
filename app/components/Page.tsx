import colors from "@assets/colors";
import { StyleSheet, View, ViewProps } from "react-native";

export interface PageProps extends ViewProps {}

const Page = ({ ...props }: PageProps) => {
  return <View {...props} style={styles.container} />;
};

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    backgroundColor: colors.background,
  },
});

export default Page;
