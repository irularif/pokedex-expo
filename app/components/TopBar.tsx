import {
  StatusBar,
  StatusBarProps,
  StyleSheet,
  View,
  ViewProps,
} from "react-native";
import { useSafeAreaInsets } from "react-native-safe-area-context";
import colors from "@assets/colors";
import { getStyleValue } from "@app/helpers";

export interface TopBarProps extends ViewProps {
  statusBarProps?: StatusBarProps;
}

const TopBar = ({ statusBarProps, children, style, ...props }: TopBarProps) => {
  // add padding top for notch with safe area insets
  const inset = useSafeAreaInsets();
  const topbarStyle = StyleSheet.flatten([styles.container, style]);
  const finalTopBarStyle = StyleSheet.flatten([
    topbarStyle,
    {
      paddingTop:
        getStyleValue(
          topbarStyle,
          ["padding", "paddingVertical", "paddingTop"],
          0
        ) + inset.top,
    },
  ]);

  return (
    <View {...props} style={finalTopBarStyle}>
      <StatusBar
        backgroundColor="transparent"
        translucent={true}
        barStyle="dark-content"
        {...statusBarProps}
      />
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    paddingHorizontal: 16,
    paddingVertical: 8,
  },
  title: {
    color: colors.white,
    fontWeight: "600",
    fontSize: 20,
  },
});

export { styles as topBarStyles };
export default TopBar;
