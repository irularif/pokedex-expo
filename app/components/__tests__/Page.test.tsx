import React from "react";
import { Text } from "react-native";
import { renderWithWrapper } from "../../../.ci/testHelper";
import Page from "../Page";
import TopBar from "../TopBar";

describe("Page Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Page>
        <TopBar>
          <Text>Pok√©dex</Text>
        </TopBar>
      </Page>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
