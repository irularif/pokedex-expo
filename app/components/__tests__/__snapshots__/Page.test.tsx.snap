// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Component should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
